nwb_conversion_tools/datainterfaces/behavior/movie/movie_utils.py

```diff
@@ -13,6 +15,134 @@
 PathType = Union[str, Path]
 
 
+class VideoCaptureContext:
+    def __init__(self, *args, stub=False, **kwargs):
+        self.vc = cv2.VideoCapture(*args, **kwargs)
+        self._args = args
+        self._kwargs = kwargs
+        self.stub = stub
+        self._current_frame = 0
+        self.frame_count = self.get_movie_frame_count()
+        assert self.frame_count > 0, "movie contains no frames"
+        self.fps = self.get_movie_fps()
+        self.frame = self.get_movie_frame(0)
+        assert self.frame is not None, "unable to read the movie file provided"
+
+    def get_movie_timestamps(self):
+        """
+        Return numpy array of the timestamps for a movie file.
+
+        """
+        if not self.vc.isOpened():
+            raise ValueError("movie file is not open")
+        ts = [self.vc.get(cv2.CAP_PROP_POS_MSEC)]
+        for i in tqdm(range(1, self.get_movie_frame_count()), desc="retrieving video timestamps"):
+            self.current_frame = i
+            ts.append(self.vc.get(cv2.CAP_PROP_POS_MSEC))
+        self.current_frame = 0
+        return np.array(ts)
+
+    def get_movie_fps(self):
+        """
+        Return the internal frames per second (fps) for a movie file.
+
+        """
+        if int(cv2.__version__.split(".")[0]) < 3:
+            return self.vc.get(cv2.cv.CV_CAP_PROP_FPS)
+        return self.vc.get(cv2.CAP_PROP_FPS)
+
+    def get_frame_shape(self) -> Tuple:
+        """
+        Return the shape of frames from a movie file.
+        """
+        return self.frame.shape
+
+    def get_movie_frame_count(self):
+        """
+        Return the total number of frames for a movie file.
+
+        """
+        if self.stub:
+            # if stub the assume a max frame count of 10
+            return 10
+        if int(cv2.__version__.split(".")[0]) < 3:
+            count = self.vc.get(cv2.cv.CV_CAP_PROP_FRAME_COUNT)
+        else:
+            count = self.vc.get(cv2.CAP_PROP_FRAME_COUNT)
+        return int(count)
```

Comment:
```suggestion
        if self.vc.isOpened():
            if int(cv2.__version__.split(".")[0]) < 3:
                count = self.vc.get(cv2.cv.CV_CAP_PROP_FRAME_COUNT)
            else:
                count = self.vc.get(cv2.CAP_PROP_FRAME_COUNT)
            return int(count)
```

This could make access to this field even safer.